File: hw/nase/u3/hw23AbstractFactory/AbstractFactory.JavaScript/controller/ClientAppController.js
import { GUIFactory } from '../model/GUIFactory.js';
import { Console } from '../view/Console.js';

export class ClientAppController {
    static async run() {
        const view = new Console();
        
        const a_factory = await GUIFactory.getFactory();
        
        const a_button = a_factory.createButton();
        a_button.caption = "Play";
        view.renderButton(a_button);
        
        const a_menu = a_factory.createMenu();
        a_menu.caption = "File";
        view.renderMenu(a_menu);
    }
}

File: hw/nase/u3/hw23AbstractFactory/AbstractFactory.python/controller/ClientAppController.py
from model.GUIFactory import GUIFactory
from view.Console import Console

class ClientAppController:
    @staticmethod
    def run():
        view = Console()
        
        a_factory = GUIFactory.get_factory()
        
        a_button = a_factory.create_button()
        a_button.caption = "Play"
        view.render_button(a_button)
        
        a_menu = a_factory.create_menu()
        a_menu.caption = "File"
        view.render_menu(a_menu)